gensrc/proto/internal_service.proto

```diff
@@ -740,6 +740,26 @@ message PUpdateTransactionStateResponse {
     repeated StatusPB results = 1;
 }
 
+message PColumn {
+    optional int32 slot_id = 1;
+    optional int64 data_size = 2;
+    optional bytes data = 3;
+}
+
+message PLookUpRequest {
+    optional PUniqueId query_id = 1;
+    optional int32 lookup_node_id = 2;
+    optional int32 request_tuple_id = 3;
+    repeated PColumn request_columns = 4;
+    repeated PSlotDescriptor lookup_slots = 5;
+}
+
+message PLookUpResponse {
+    optional StatusPB status = 1;
+    repeated PColumn columns = 2;
```

Comment:
can we use column-oriented compression or encoding/decoding to minimize both requests and responses?

Reply:
yes, I will support it in later pr.